file/src/main/java/org/apache/zeppelin/file/FileInterpreter.java

```diff
@@ -0,0 +1,170 @@
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.file;
+
+import org.apache.zeppelin.interpreter.Interpreter;
+import org.apache.zeppelin.interpreter.InterpreterContext;
+import org.apache.zeppelin.interpreter.InterpreterResult;
+import org.apache.zeppelin.interpreter.InterpreterResult.Code;
+import org.apache.zeppelin.interpreter.InterpreterResult.Type;
+import org.apache.zeppelin.scheduler.Scheduler;
+import org.apache.zeppelin.scheduler.SchedulerFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+
+/**
+ * File interpreter for Zeppelin.
+ *
+ * @author rajbains
+ */
+public abstract class FileInterpreter extends Interpreter {
+  Logger logger = LoggerFactory.getLogger(FileInterpreter.class);
+  String currentDir = null;
+  CommandArgs args = null;
+
+  public FileInterpreter(Properties property) {
+    super(property);
+    currentDir = new String("/");
+  }
+
+  /**
+   * Handling the arguments of the command
+   */
+  public class CommandArgs {
+    public String input = null;
+    public String command = null;
+    public ArrayList<String> args = null;
+    public HashSet<Character> flags = null;
+
+    public CommandArgs(String cmd) {
+      input = cmd;
+      args = new ArrayList();
+      flags = new HashSet();
+    }
+
+    private void parseArg(String arg) {
+      if (arg.charAt(0) == '-') {                   // handle flags
+        for (int i = 0; i < arg.length(); i++) {
+          Character c = arg.charAt(i);
+          flags.add(c);
+        }
+      } else {                                      // handle other args
+        args.add(arg);
+      }
+    }
+
+    public void parseArgs() {
+      if (input == null)
+        return;
+      StringTokenizer st = new StringTokenizer(input);
+      if (st.hasMoreTokens()) {
+        command = st.nextToken();
+        while (st.hasMoreTokens())
+          parseArg(st.nextToken());
+      }
+    }
+  }
+
+  // Functions that each file system implementation must override
+
+  public abstract String listAll(String path);
+
+  public abstract boolean isDirectory(String path);
+
+  // Combine paths, takes care of arguments such as ..
+
+  private String getNewPath(String argument){
+    Path arg = Paths.get(argument);
+    Path ret = arg.isAbsolute() ? arg : Paths.get(currentDir, argument);
+    return ret.normalize().toString();
+  }
+
+  // Handle the command handling uniformly across all file systems
+
+  @Override
+  public InterpreterResult interpret(String cmd, InterpreterContext contextInterpreter) {
+    logger.info("Run File command '" + cmd + "'");
+
+    args = new CommandArgs(cmd);
+    args.parseArgs();
+
+    if (args.command == null) {
+      logger.info("Error: No command");
+      return new InterpreterResult(Code.ERROR, Type.TEXT, "No command");
+    }
+
+    // Simple parsing of the command
+
+    if (args.command.equals("cd")) {
```

Comment:
The current commands are not specific to HDFS - they're ls, cd and pwd, something that you'll use in any linux shell. It might be better to use the same general commands so that the user does not have to learn new commands for each file system. If a file system needs something specific, they can override and still call the base class for the shared functionality. This way parsing is also handled in one place.